protypes/Eclipse GUI with Wizards and Dialogs/src/MyPageTwo2.java

```diff
@@ -0,0 +1,100 @@
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.KeyEvent;
+import org.eclipse.swt.events.KeyListener;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+
+public class MyPageTwo2 extends WizardPage {
+
+	private Text text1;
+
+	private Composite container;
+
+	public MyPageTwo2() {
+
+		super("Second Page");
+
+		setTitle("Second Page");
+
+		setDescription("Now this is the second page");
+
+		setControl(text1);
+
+	}
+
+	@Override
+
+	public void createControl(Composite parent) {
+
+		container = new Composite(parent, SWT.NONE);
+
+		GridLayout layout = new GridLayout();
+
+		container.setLayout(layout);
+
+		layout.numColumns = 2;
+
+		Label label1 = new Label(container, SWT.NONE);
+
+		label1.setText("Say hello to Fred");
+
+		text1 = new Text(container, SWT.BORDER | SWT.SINGLE);
+
+		text1.setText("");
+
+		text1.addKeyListener(new KeyListener() {
+
+			@Override
+
+			public void keyPressed(KeyEvent e) {
+
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+
+			public void keyReleased(KeyEvent e) {
+
+				if (!text1.getText().isEmpty()) {
+
+					setPageComplete(true);
+
+				}
+
+			}
+
+		});
+
+		GridData gd = new GridData(GridData.FILL_HORIZONTAL);
+
+		text1.setLayoutData(gd);
+
+		Label labelCheck = new Label(container, SWT.NONE);
+
+		labelCheck.setText("This is a check");
+
+		Button check = new Button(container, SWT.CHECK);
+
+		check.setSelection(true);
+
+		// required to avoid an error in the system
+
+		setControl(container);
+
+		setPageComplete(false);
+
+	}
+
+	public String getText1() {
+
+		return text1.getText();
+
+	}
+
+}
```

Comment:
When does this page even appear? I can’t remember to see it when I tried out your code.